File: src/frontend/src/types/document.ts
import { DocumentType, DocumentStatus, AmountConsistencyStatus } from './enums'

export interface Attachment {
  id: string
  documentId: string
  fileName: string
  originalFileName: string
  filePath: string
  contentType: string
  fileSize: number
  description?: string
  uploadedById: string
  uploadedByName?: string
  createdAt: string
}

export interface DocumentItem {
  id: string
  itemOrder: number
  itemCode: string
  name: string
  specification?: string
  unit: string
  quantity: number
  unitPrice: number
  subtotal: number
  notes?: string
  materialId?: string
  materialName?: string
}

export interface Document {
  id: string
  documentNumber: string
  type: DocumentType
  title: string
  description?: string
  expectedAmount: number
  actualAmount?: number
  amountDifference: number
  amountConsistency: AmountConsistencyStatus
  status: DocumentStatus
  measurementDate: string
  approvalDate?: string
  createdAt: string
  projectName: string
  projectId: string
  createdByName: string
  createdById: string
  itemCount: number
  attachmentCount: number
  pendingApprovalCount: number
  items?: DocumentItem[]
  history?: DocumentHistory[]
}

export interface CreateDocumentItemDto {
  itemOrder: number
  itemCode: string
  name: string
  specification?: string
  unit: string
  quantity: number
  unitPrice: number
  notes?: string
  materialId?: string
}

export interface UpdateDocumentItemDto {
  id?: string
  itemOrder: number
  itemCode: string
  name: string
  specification?: string
  unit: string
  quantity: number
  unitPrice: number
  notes?: string
  materialId?: string
}

export interface CreateDocumentDto {
  type: DocumentType
  title: string
  description?: string
  expectedAmount: number
  measurementDate: string
  projectId: string
  items: CreateDocumentItemDto[]
}

export interface UpdateDocumentDto {
  title: string
  description?: string
  expectedAmount: number
  actualAmount?: number
  measurementDate: string
  status: DocumentStatus
  items: UpdateDocumentItemDto[]
}

export interface BatchUpdateDocumentsDto {
  documentIds: string[]
  status?: DocumentStatus
  assignedToId?: string
}

export interface DocumentHistory {
  id: string
  action: string
  oldValues?: string
  newValues?: string
  materialsBefore?: string
  materialsAfter?: string
  conclusion?: string
  source?: string
  oldStatus?: DocumentStatus
  newStatus?: DocumentStatus
  createdAt: string
  createdByName: string
}

export interface DocumentFilterParams {
  type?: DocumentType
  status?: DocumentStatus
  amountConsistency?: AmountConsistencyStatus
  projectId?: string
  startDate?: string
  endDate?: string
  search?: string
}

File: src/backend/HomeImprovementPlatform.API/Models/Document.cs
using HomeImprovementPlatform.API.Enums;

namespace HomeImprovementPlatform.API.Models;

public class Document
{
    public Guid Id { get; set; }
    public string DocumentNumber { get; set; } = string.Empty;
    public DocumentType Type { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public decimal ExpectedAmount { get; set; }
    public decimal? ActualAmount { get; set; }
    public AmountConsistencyStatus AmountConsistency { get; set; } = AmountConsistencyStatus.PendingVerification;
    public DocumentStatus Status { get; set; } = DocumentStatus.Draft;
    public DateTime MeasurementDate { get; set; }
    public DateTime? ApprovalDate { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }

    public Guid ProjectId { get; set; }
    public virtual Project? Project { get; set; }
    public Guid CreatedById { get; set; }
    public virtual ApplicationUser? CreatedBy { get; set; }

    public virtual ICollection<DocumentItem> Items { get; set; } = new List<DocumentItem>();
    public virtual ICollection<ApprovalNode> ApprovalNodes { get; set; } = new List<ApprovalNode>();
    public virtual ICollection<Attachment> Attachments { get; set; } = new List<Attachment>();
    public virtual ICollection<DocumentHistory> DocumentHistories { get; set; } = new List<DocumentHistory>();
}
